compiler/src/iree/compiler/Dialect/LinalgExt/Transforms/RewriteFft.cpp

```diff
@@ -0,0 +1,132 @@
+// Copyright 2024 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//
+
+#include "iree/compiler/Dialect/LinalgExt/IR/LinalgExtOps.h"
+#include "iree/compiler/Dialect/LinalgExt/Transforms/Transforms.h"
+
+namespace mlir::iree_compiler::IREE::LinalgExt {
+
+FailureOr<std::pair<Value, Value>> rewriteFft(Operation *op, Value operand,
+                                              int64_t fftLength,
+                                              PatternRewriter &rewriter) {
+
+  assert(llvm::isPowerOf2_64(fftLength) &&
+         "expected FFT length to be a power of two");
+  Location loc = op->getLoc();
+
+  auto operandType = llvm::dyn_cast<RankedTensorType>(operand.getType());
+  if (!operandType || !operandType.hasStaticShape()) {
+    return failure();
+  }
+
+  // Skip else getBitReversalOrder produces invalid dense elements attr.
+  if (isa<ComplexType>(getElementTypeOrSelf(operand.getType())))
+    return rewriter.notifyMatchFailure(op, "expected real types");
+
+  ImplicitLocOpBuilder b(loc, rewriter);
+
+  auto getBitReversalOrder = [](ImplicitLocOpBuilder &b, Value real,
+                                int64_t fftLength) -> SmallVector<Value> {
+    auto realType = llvm::cast<ShapedType>(real.getType());
+    auto rank = realType.getRank();
```

Comment:
nit: spell out full type
```suggestion
    int64_t rank = realType.getRank();
```

Reply:
Applying.